crates/blockifier/tests/erc20_tests.rs

```diff
@@ -1,165 +1,577 @@
+pub const U256_UNDERFLOW: &str = "0x753235365f737562204f766572666c6f77";
+pub const NAME: &str = "Native";
+pub const SYMBOL: &str = "MTK";
+pub const DECIMALS: u128 = 18;
+
+#[test]
+fn should_deploy() {
+    let (_contract_address, _state) = prepare_erc20_deploy_test_state();
 }
 
-fn deploy_contract(
-    state: &mut dyn State,
-    class_hash: Felt,
-    contract_address_salt: Felt,
-    calldata: &[Felt],
-) -> SyscallResult<(Felt, Vec<Felt>)> {
-    let deployer_address = ContractAddress::default();
-
-    let class_hash = ClassHash(felt_to_starkfelt(class_hash));
-
-    let wrapper_calldata = Calldata(Arc::new(
-        calldata.iter().map(|felt| felt_to_starkfelt(*felt)).collect::<Vec<StarkFelt>>(),
-    ));
-
-    let calculated_contract_address = calculate_contract_address(
-        ContractAddressSalt(felt_to_starkfelt(contract_address_salt)),
-        class_hash,
-        &wrapper_calldata,
-        deployer_address,
-    )
-    .map_err(|_| vec![Felt::from_hex(FAILED_TO_CALCULATE_CONTRACT_ADDRESS).unwrap()])?;
-
-    let ctor_context = ConstructorContext {
-        class_hash,
-        code_address: Some(calculated_contract_address),
-        storage_address: calculated_contract_address,
-        caller_address: deployer_address,
-    };
-
-    let call_info = execute_deployment(
-        state,
-        &mut Default::default(),
-        &mut EntryPointExecutionContext::new(
-            &BlockContext::create_for_testing(),
-            &AccountTransactionContext::Current(Default::default()),
-            ExecutionMode::Execute,
-            false,
-        )
-        .unwrap(),
-        ctor_context,
-        wrapper_calldata,
-        u64::MAX,
-    )
-    .map_err(|_| vec![Felt::from_hex(FAILED_TO_EXECUTE_CALL).unwrap()])?;
-
-    let return_data =
-        call_info.execution.retdata.0[..].iter().map(|felt| starkfelt_to_felt(*felt)).collect();
-
-    let contract_address_felt =
-        Felt::from_bytes_be_slice(calculated_contract_address.0.key().bytes());
-
-    Ok((contract_address_felt, return_data))
+#[cfg(test)]
+mod read_only_methods_tests {
+    use super::*;
+
+    #[test]
+    fn test_total_supply() {
+        let mut context = TestContext::new();
+
+        let result = context.call_entry_point("total_supply", vec![]);
+
+        assert_eq!(result, vec![Felt::from(TOTAL_SUPPLY), Felt::from(0u8)]);
+    }
+
+    #[test]
+    fn test_balance_of() {
+        let address = felt_to_starkfelt(contract_address_to_felt(Signers::Alice.into()));
+
+        let mut context = TestContext::new();
+        let result = context.call_entry_point("balance_of", vec![address]);
+
+        assert_eq!(result, vec![Felt::from(TOTAL_SUPPLY), Felt::from(0u8)]);
+    }
 }
 
-pub fn prepare_erc20_deploy_test_state() -> (ContractAddress, CachedState<DictStateReader>) {
-    let mut state = create_erc20_deploy_test_state();
+mod transfer_tests {
+    use super::*;
+
+    #[test]
+    fn test_transfer_normal_scenario() {
```

Comment:
There is something I don't understand here. The contract we are compiling doesn't have a transfer function, how are you calling it?

Reply:
It's part of erc20 core functionality 
```
#[starknet::interface]
trait IERC20<TState> {
    fn total_supply(self: @TState) -> u256;
    fn balance_of(self: @TState, account: ContractAddress) -> u256;
    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
    fn transfer_from(
        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
    ) -> bool;
    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
}
```